File: apps/java/io/dustin/apps/board/domain/restriction/posting/service/ReadPostingRestrictionService.java
package io.dustin.apps.board.domain.restriction.posting.service;


import io.dustin.apps.board.domain.restriction.posting.model.PostingRestriction;
import io.dustin.apps.board.domain.restriction.posting.repository.PostingRestrictionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import static io.dustin.apps.common.utils.OptionalUtils.getEntity;

@RequiredArgsConstructor
@Service
public class ReadPostingRestrictionService {

    private final PostingRestrictionRepository postingRestrictionRepository;

    public PostingRestriction getPostingRestrictionUser(Long fromUserId, Long toUserId, Long postingId) {
        return getEntity(postingRestrictionRepository.findByFromUserIdAndToUserIdAndPostingId(fromUserId, toUserId, postingId), PostingRestriction.class, "data not found");
    }

}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/restriction/posting/service/ReadPostingRestrictionService.kt
package io.dustin.apps.board.domain.restriction.posting.service

import io.dustin.apps.board.domain.restriction.posting.model.PostingRestriction
import io.dustin.apps.board.domain.restriction.posting.repository.PostingRestrictionRepository
import io.dustin.apps.common.utils.notFoundEntity
import org.springframework.stereotype.Service

@Service
class ReadPostingRestrictionService (
    private val postingRestrictionRepository: PostingRestrictionRepository
) {

    /**
     * 제한된 유저 객체 가져오기
     */
    fun getPostingRestrictionUser(fromUserId: Long, toUserId: Long, postingId: Long): PostingRestriction {
        val errorMessage = "조회된 정보가 없습니다."
        return postingRestrictionRepository.findByUsers(fromUserId, toUserId, postingId) ?: notFoundEntity(errorMessage)
    }

}
